### DISCUSSION TITLE
Would you have a plan to add more format options like Vetur?

### DISCUSSION THREAD

**Discussion by zheung** [vuejs/language-tools#444] [Ideas]
Volar's language support makes it a pleasure to use. As a heavy user of formatting code, Volar is very scary for me. Not only did it force format my `<template>` code, but it also did not provide `initial indent` settings. It doesn't even offer the option to close the formatter.  
  
I really hate it when plugins make changes to my code that are not expected. That's why I still can't use Volar in my daily programming. I would like to know your plans for the relevant options: )

![Vetur Setting 1](https://user-images.githubusercontent.com/894004/131596794-8bf1c627-6aa6-4de2-8458-c19695e26d34.png)
![Vetur Setting 2](https://user-images.githubusercontent.com/894004/131596799-c72e5a85-003e-45a3-b27d-a99e4d75680c.png)

**Comment by johnsoncodehk** [comment]
Volar formatter is designed to be used out of the box, for more custom rules I recommend switching to Prettier.

Volar don't force formatting your code, unless you set automatic formatting in VSCode, please check your VSCode settings.

See https://github.com/johnsoncodehk/volar/discussions/53, https://github.com/johnsoncodehk/volar/issues/425.

> but it also did not provide `initial indent` settings. It doesn't even offer the option to close the formatter.

You can open a feature request for this.

**Reply by zheung** [reply]
Yep, Volar doesn't force formatting my code. I mean I just naturally use its default Formatter : )  
  
Sadly, only the configured Vetur works for my needs, not even Prettier. neither Prettier nor Volar offers a switch to format different languages independently. And Volar and Vetur conflict : (

I will try to open a feature request for an independent formatting switch. Thank you for your answer

**Reply by johnsoncodehk** [reply]
You can still enable Vetur and Volar together, and select Vetur as the default formatter, but you may want to turn off features other than format in the workspace Vetur settings.

**Reply by zheung** [reply]
Yes this is what I want. I tried enabling them both and select Vetur as the default formatted last month. They seem to be conflicting on Language Support. But I just tried again and it shows everything is fine. This gives me hope 😋

**Reply by mesqueeb** [reply]
@zheung I need something similar, I need `sass-formatter` to be used as the formatter for the `<style lang="sass>` sections of Vue files. Can you share your current setup to make it work?

I'm also trying to switch from Vetur. 😅 

One more question: I think both Vetur and Volar use `prettier` for the `<template>` section of Vue files, but it seems that there are still differences where Volar formats differently from Vetur. Do you know the origin of this or what causes it?

Thanks a lot!

**Reply by SomeoneIsWorking** [reply]
Prettier is not configurable at all, as it says "opinionated"

It turns

    [{
        a: 5,
        b: 4
    }, {
        a: 4,
        b: 5
    }]

Into 

    [
        {
            a: 5,
            b: 4
        }, 
        {
            a: 4,
            b: 5
        }
    ]

(Wrote from memory so commas and indentation might be different)
IMO should be stayed away as far away as possible

> Yes this is what I want. I tried enabling them both and select Vetur as the default formatted last month. They seem to be conflicting on Language Support. But I just tried again and it shows everything is fine. This gives me hope 😋

Interesting. I couldn't resolve conflicts but I'm going to try again.

**Reply by douglasg14b** [reply]
Yeah, double that, prettier has abysmal configuration.
